src/Components/test/testassets/BasicTestApp/RouterTest/TestRouterWithLazyAssembly.razor

```diff
@@ -0,0 +1,74 @@
+@using Microsoft.AspNetCore.Components.Routing
+@using Microsoft.AspNetCore.Components.WebAssembly
+@using System.Linq
+@using System.Reflection
+
+@inject WebAssemblyLazyAssemblyLoader lazyLoader
+
+<Router AppAssembly="@typeof(BasicTestApp.Program).Assembly" AdditionalAssemblies="@lazyLoadedAssemblies" OnNavigate="@OnNavigate">
+    <Found Context="routeData">
+        @if (loading) {
+            <div style="padding: 20px;background-color:blue;color:white;" id="loading-banner" @onclick="HideLoadingBar">
```

Comment:
This may be a work-in-progress, so please ignore me if I’m commenting too soon, but it seems odd that the user would manually click this to exit loading state.

Reply:
Or maybe you’re just doing this for E2E automation purposes. Totally fine if so, but I’d still like to see how it would be done for real. 

Presumably it’s not enough just to call this after awaiting the load, because there would be a race condition if the user navigates multiple times while loading is in progress.

Reply:
I think you might be right. 

We can probably handle this internally within the assembly loader. The loader can keep track of the things that it is loading, fire up requests for new things to load, but then return a task that only completes after all the previous requests have completed sequentially. It's a bit involved but it's all internal to the loader. That said, let's discuss this before we decide to do it.

Reply:
We can also use a CTS internally to cancel the previous tasks if the loader receives a new request, (or have something on top that essentially does that)